core/org.wso2.carbon.utils/src/main/java/org/wso2/carbon/utils/dbcreator/DatabaseCreator.java

```diff
@@ -349,6 +350,21 @@ private void executeSQLScript() throws Exception {
         }
     }
 
+    private String getDatabaseTypeWithVersion(String databaseType) throws SQLException {
+
+        if (databaseType != null && databaseType.equals("mysql")) {
+            if (conn != null && (!conn.isClosed())) {
+                DatabaseMetaData metaData = conn.getMetaData();
+                int databaseMajorVersion = metaData.getDatabaseMajorVersion();
+                int databaseMinorVersion = metaData.getDatabaseMinorVersion();
+                if (databaseMajorVersion == 5 && databaseMinorVersion >= 7) {
+                    databaseType = databaseType + "-5.7+";

```

Comment:
We need to switch the database script based on the variable explicit_defaults_for_timestamp of MySQL, not the version. The only impact of MySQL version here is they have changed the default value of the variable in 5.7.